zingstats/changes.py

```diff
@@ -0,0 +1,287 @@
+#
+# (c) Copyright 2018 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import base64
+import json
+import logging
+import urllib
+from datetime import datetime
+
+import requests
+
+log = logging.getLogger(__name__)
+
+
+class Message(object):
+    def __init__(self, message_id, message_date, message_text):
+        self.message_id = message_id
+        self.message_dt = None
+        self.text = message_text
+
+
+class GerritMessage(Message):
+    def __init__(self, message_id, message_date, message_text):
+        super(GerritMessage, self).__init__(message_id, message_date,
+                                            message_text)
+        self.message_id = message_id
+        self._message_ts = message_date[:-3]
+        self.message_dt = datetime.strptime(self._message_ts,
+                                            GerritChange.GERRIT_FORMAT)
+
+
+class Revision(object):
```

Comment:
It's not clear to me what a Revision object is without reading through the higer level code.